src/vm/field.h

```diff
@@ -85,10 +85,35 @@ class FieldDesc
     LPUTF8 m_debugName;
 #endif
 
+public:
     // Allocated by special heap means, don't construct me
-    FieldDesc() {};
+    FieldDesc() =delete;
+
+#ifndef DACCESS_COMPILE
+    FieldDesc& operator = (const FieldDesc& src)
```

Comment:
Is it used in `src/vm/generics.cpp`:
```
if (pOldMT->GetNumStaticFields() != 0)
{
    pStaticFieldDescs = (FieldDesc*) pamTracker->Track(pAllocator->GetLowFrequencyHeap()->AllocMem(S_SIZE_T(sizeof(FieldDesc)) * S_SIZE_T(pOldMT->GetNumStaticFields())));
    FieldDesc* pOldFD = pOldMT->GetGenericsStaticFieldDescs();

    g_IBCLogger.LogFieldDescsAccess(pOldFD);

    for (DWORD i = 0; i < pOldMT->GetNumStaticFields(); i++)
    {
        pStaticFieldDescs[i] = pOldFD[i];
        pStaticFieldDescs[i].SetMethodTable(pMT);
    }
}
```

The pointer in is overwritten `SetMethodTable` just after copy assignment.

Is it better to replace the assignment operator with a `Copy` or `InitializeFrom` method?